src/service/resources-impl.js

```diff
@@ -506,7 +508,8 @@ export class Resources {
     const parentResource = Resource.forElement(parentElement);
     subElements = elements_(subElements);
 
-    this.discoverResourcesForArray_(parentResource, subElements, resource => {
+    this.discoverResourcesForArray_(parentResource, subElements,
+    /* deepSubtree */ true, resource => {
```

Comment:
So, I'd say, `deepSubtree` should not go into `owned` elements.

That being said, I'd like to discuss this offline - I think we just need to go ahead and start working on layers. Possibly we can squeeze this PR in for a temporary fix, but I'd like to make sure we are not digging a deeper hole.